### PR TITLE
Refactor worker for high available DM

### PR DESCRIPTION
**Pull request by Little-Wallace**
### What problem does this PR solve? 

Make master manage the config of mysql.

### Check List 

Tests 

 - Unit test

Code changes

- master: If you want to start some task which would subscribe data from mysql, you must create a mysql task in master at first.
```
func OperateMysqlWorker(WorkerConfigRequest)WorkerConfigResponse // create mysql relay log worker / update mysql config / stop mysql worker.
```
- worker: If you want worker to do some task which sync data from mysql to TiDB, you must start a mysql task in this server at first.

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

### CODE REVIEW

File: dm/worker/server.go [hunk 1]
```diff
@@ -75,23 +77,41 @@ func (s *Server) Start() error {
...
+				if shouldStop {
+					s.Lock()
+					if s.worker != nil {
+						s.worker.Close()
+						s.worker = nil
+					}
+					s.Unlock()
+					shouldStop = false
+				} else {
+					// Try to connect master again before stop worker
```
Comment (csuzhangxc) [inline dm/worker/server.go]: Are `TODO` for this line and L110?

File: dm/proto/dmworker.proto [hunk 2]
```diff
@@ -448,3 +454,32 @@ message QueryWorkerConfigResponse {
...
+}
+
+message TaskLog {
+    int64 id = 1;
+    TaskMeta task = 2;
+    int64 ts = 3;
+    // true means this log is applied successfully
+    bool success = 4;
+    string message = 5;
+}
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply dm/proto/dmworker.proto]: Sorry. This code was merged into together automatically. I will remove them.

File: dm/proto/dmworker.proto [hunk 3]
```diff
@@ -39,6 +39,12 @@ service Worker {
     rpc QueryWorkerConfig (QueryWorkerConfigRequest) returns (QueryWorkerConfigResponse) {}
 
     rpc MigrateRelay(MigrateRelayRequest) returns (CommonWorkerResponse) {}
+
+    rpc CreateMysqlTask(MysqlTaskRequest) returns (MysqlTaskResponse) {}
+
+    rpc StopMysqlTask(StopMysqlTaskRequest) returns (MysqlTaskResponse) {}
+
+    rpc UpdateMysqlTaskConfig(MysqlTaskRequest) returns (MysqlTaskResponse) {}
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply dm/proto/dmworker.proto]: ok, it's a good idea.

File: dm/master/coordinator/worker.go [hunk 4]
```diff
@@ -81,6 +85,64 @@ func (w *Worker) State() WorkerState {
...
+	if err != nil {
+		return nil, err
+	}
+	ownerReq := &workerrpc.Request{
+		Type: workerrpc.CmdUpdateMysqlConfig,
+		MysqlWorkerRequest: &pb.MysqlTaskRequest{
+			Config: content,
+		},
+	}
+	resp, err := w.client.SendRequest(ctx, ownerReq, d)
```
[the remark this answers is not in the document]
Comment (Little-Wallace) [reply dm/master/coordinator/worker.go]: Yes. I should use 'GetClient()' instead.

File: dm/worker/server.go (hunk 1 above)
Comment (Little-Wallace) [reply dm/worker/server.go]: Why? This gorutine  will keep trying KeepAlive util it receives a signal which set 'shouldExit' true.

**Pull request merged**